Library/Homebrew/extend/os/linux/hardware/cpu.rb

```diff
@@ -3,17 +3,17 @@
 module Hardware
   class CPU
     class << self
-      OPTIMIZATION_FLAGS_LINUX = {
-        native:  "-march=#{Homebrew::EnvConfig.arch}",
-        nehalem: "-march=nehalem",
-        core2:   "-march=core2",
-        core:    "-march=prescott",
-        armv6:   "-march=armv6",
-        armv8:   "-march=armv8-a",
-      }.freeze
-
       def optimization_flags
-        OPTIMIZATION_FLAGS_LINUX
+        {
+          native:  arch_flag(Homebrew::EnvConfig.arch),
+          nehalem: "-march=nehalem",
+          core2:   "-march=core2",
+          core:    "-march=prescott",
+          armv6:   "-march=armv6",
+          armv8:   "-march=armv8-a",
+          ppc64:   "-mcpu=powerpc64",
+          ppc64le: "-mcpu=powerpc64le",
+        }
```

Comment:
```suggestion
        @optimization_flags ||= begin
          flags = generic_optimization_flags.dup
          flags[:native] = arch_flag(Homebrew::EnvConfig.arch)
          flags
        end
```

maybe something like this? Haven't checked that it works or is correct!